### ISSUE TITLE
Non-blocking/asynchronous behavior of script.reset()

### ISSUE THREAD

**Issue by dolandemort** [Blightmud/Blightmud#601]
I have a reload function. It resets the lua environment and then loads the main config script, with the capability of reloading server specific scripts (in different folder) if the mud is currently connected to:

``` lua
local function reload_lua(state)
  script.reset()
  script.load("/some/path/config.lua")

  blight.status_height(1)
  blight.status_line(0, "")

  if (state ~= "partial" and store.session_read("connected") == "1") then
    load_server_scripts()
  end
end`
```

I am loading server specific scripts using `dofile`, but there is a problem arising from the nature of `script.reset()`: Even though it is called before loading the server scripts, the server scripts are first loaded, and, at some point after that, the call of `script.reset()` finishes, resetting the lua state.

**Comment by dolandemort** [comment]
The "file loading callbacks" sounds interesting and useful. Where can I read/learn about them, if you don't mind me asking.

**Comment by LiquidityC** [comment]
`/help script` if my memory serves me.

**Comment by dolandemort** [comment]
Additionally, I have started to explore the other alternative you mentioned, namely `on_connect` firing if you call `script.reset()`. Unless I am doing something foolish, it doesn't appear to function like that.

I added a simple `print('debug')` to my `on_connect` callback. I connect to a mud and I, as expected see my 'debug' output. I then call my reload function and there is no additional 'debug' output resulting.

**Comment by LiquidityC** [comment]
Poor choice of words. I mean the `script.load()` function. It's also async but all async functions are guaranteed to execute in the order they are called.

**Comment by LiquidityC** [comment]
You are right about the on_connect. I'm remembering an older implementation. `/help config_scripts` has an example which mimics that behavior. Also `script.on_reset` could solve your problem as well. You need to use `script.load` when loading the root file of your script. Everything beyond that you can load as you like.

**Comment by dolandemort** [comment]
Alright. The `config_scripts` help file with the code approximating the functionality of the previous version seems to be what I needed. Appreciate all the help.
